File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        pointer = 1
        for i,v in enumerate(numbers):
            while pointer < len(numbers):
                if (v + numbers[pointer]) == target:
                    return [i+1,pointer+1]
                pointer += 1
            pointer = i+1

        

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> indices;

        for (int i = 0; i < nums.size(); i ++){
            int needed = target - nums[i];

            if (indices.count(needed)){
                return {indices[needed],i};
            }
            indices[nums[i]] = i;
        }
        return {};
    }
};
